NAAM.md

```diff
@@ -0,0 +1,275 @@
+
+Once ingested, indexer nodes will eventually remove the Naam advertisement metadata along with the
+publisher information.
+
+### Resolving IPNS Records
+
+IPNS record resolution involves finding the IPNS record associated to a given IPNS key, i.e.
+`/ipns/<ipns-key>`. As eluded to earlier, Naam uses the find API already provided by the network
+indexers to achieve this. To resolve an IPNS record, a resolver:
+
+* calculates the multihash of the given IPNS key as its SHA-256,
+* looks up the provider info for that multihash via the IPNI find API,
+* decodes the metadata from the provider info as IPNS record,
+* validates the record, and
+* if valid returns its value.
+
+## Limitations
+
+### IPNS Record Size
+
+The maximum size of IPNS records supported by Naam is 1 KiB, even though IPNS specification suggests
+a maximum limit of 10 KiB. The 1 KiB limit currently imposed by network indexers on the maximum size
+of advertisement metadata.
+
+Basic tests show that a 1KiB limit is sufficiently large for storing simple IPNS records pointing to
+IPFS paths, specially for records that do not need the peer's public key to be embedded into IPNS
+records. For example, a Ed25519 peer ID also captures the peer's public key. A marshalled IPNS
+record with such key type and empty value constructed
+using [`go-ipns`](https://github.com/ipfs/go-ipns) library comes to `257` bytes, which leaves
+sufficient space for encoding fairly long IPFS paths as the record value.
+
+Because of the size limitation NAAM Records also limit storing large IDENTITY CIDs as record value.  
+
+### Record Validity Duration
+
+The indexers will maintain information from inactive providers for up to a week. This means IPNS
+records published by Naam with EOL of beyond a week should be re-published to remain resolvable.
+
+## Security
+
+### Advertisement Verifiability
+
+Naam advertisements will benefit from all the verifiability features of regular advertisements. They
+are signed by the IPNS record publisher and are verified for validity before being ingested by the
+network indexers.
+
+### Reputation
+
+Similar to the IPNI indexing protocol, the use of immutable advertisement chain by Naam provides a
+verifiable and immutable catalog of IPNS records exposed by a network participant. Access to such
+historical data enables verifiable analysis of participant behaviour, and offers an opportunity to
+design and develop reputation systems, which ultimately can contribute to the health of the network.
+As far as I know, this is something that is not currently offered by the existing implementations of
+IPNS.
+
+Even though, the specification of such reputation system is beyond the scope of this document, it is
+important to point out such potential. Because, as the size of IPFS network grows so will the need
+for reputation systems that enable users to choose from ever increasing "choices" across the
+network.
+
+## Privacy
+
+To be revisited when double-hashing work is finalised.
```

Comment:
See [go-naam PR 13](https://github.com/ipni/go-naam/pull/13).

When reader-privacy is enabled (enabled by default), a double-hashed lookup is done when resolving a key to an IPNS record. The client wishing to resolve a key to an IPNS record creates a hash of the key and sends that to the indexer for lookup. The reader-privacy-enabled indexer returns encrypted metadata that is indexed by the hash of the key.  The client then decrypts this metadata using the original key and extracts the IPNS record from the decrypted metadata.